match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -672,14 +672,17 @@ end
 
 function StarCraftMatchGroupInput.getTeamNameIcon(template)
 	local team
+	local icon
 	if template ~= nil and TeamTemplates._teamexists(template) then
-		team = TeamTemplates._team(template)
-		team = team:gsub('%&', '')
-		team = String.split(team, 'link=')[2]
-		team = String.split(team, ']]')[1]
+		team = mw.ext.TeamTemplate.raw(template)
+		icon = team.image
+		if icon == '' then
```

Comment:
Is it possible that this would require null checking?

Reply:
tested it at https://liquipedia.net/starcraft2/Module:Hjpalpha/sandbox18
it returns an empty string if the image doesn't exist (i.e. if it still uses legacy image)

Reply:
Cool, thanks :)